ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -354,6 +348,15 @@ let maybe_get_guest_metrics ~__context ~ref =
   then Some (Db.VM_guest_metrics.get_record_internal ~__context ~self:ref)
   else None
 
+(* PCI devices that belong to NVIDIA SRIOV cards *)
+let nvidia_sriov_pcis ~__context vgpus =
+  vgpus
+  |> List.filter_map (fun vgpu -> Db.VGPU.get_type ~__context ~self:vgpu
+    |> fun typ -> Db.VGPU_type.get_implementation ~__context ~self:typ
+    |> function
+      | `nvidia_sriov -> Some (Db.VGPU.get_PCI ~__context ~self:vgpu)
+      | _ -> None)
```

Comment:
If SRIOV vGPUs are now migratable should this require changes in agility.ml? Looks like we still consider all VMs with vGPUs non-agile for HA purposes, although if your pool has all compatible vGPUs they might be.

Reply:
Nevermind agility is unrelated to migration.